### PR TITLE
Corrected tooltip hover information

### LINKED ISSUE #2763: When hovering an approved string, the tooltip should also say when the strings was approved, not just by whom
Currently, it also says "Approved by X". To understand when it was reviewed, I need to look at VCS.

### PR DESCRIPTION
**Pull request by harmitgoswami**
Fix #2763 

Made changes to previous pull request to correctly display the last day an action was taken upon a translation. Additionally, removed the Weekday within the information displayed in the tooltip. See #3263 for more context.

Updated Tooltip:

Screenshot 2024-07-02 at 1 06 28 PM

### CODE REVIEW

File: pontoon/base/views.py [hunk 1]
```diff
@@ -418,7 +418,13 @@ def get_translation_history(request):
                 "date": t.date.strftime("%b %d, %Y %H:%M"),
                 "date_iso": t.date.isoformat(),
                 "approved_user": User.display_name_or_blank(t.approved_user),
+                "approved_date": t.approved_user.latest_action.created_at
+                if t.approved_user and t.approved_user.latest_action
+                else None,
```
Comment (mathjazz) [inline pontoon/base/views.py:423]: We can simply use `t.approved_date` here.

File: pontoon/base/views.py [hunk 2]
```diff
@@ -418,7 +418,13 @@ def get_translation_history(request):
...
                 "rejected_user": User.display_name_or_blank(t.rejected_user),
+                "rejected_date": t.rejected_user.latest_action.created_at
+                if t.rejected_user and t.rejected_user.latest_action
+                else None,
```
Comment (mathjazz) [inline pontoon/base/views.py:427]: Similar as above.

File: translate/public/locale/en-US/translate.ftl [hunk 3]
```diff
@@ -369,13 +369,13 @@ history-Translation--span-copied =
     .title = Copied ({ $machinerySources })
 
 history-translation--approved =
-    .title = Approved by { $user } on { DATETIME($reviewedDate, dateStyle:"long", timeStyle:"medium") }
+    .title = Approved by { $user } on { DATETIME($reviewedDate, month: "long", year: "numeric", day: "numeric", hour: "numeric", minute: "numeric", second: "numeric") }
```
Comment (mathjazz) [inline translate/public/locale/en-US/translate.ftl:372]: What's the purpose of this change?

File: translate/src/modules/history/components/HistoryTranslation.tsx [hunk 4]
```diff
@@ -196,17 +196,23 @@ export function HistoryTranslationBase({
 
   const review = {
     id: 'history-translation--unreviewed',
-    vars: { user: '', reviewedDate: new Date(translation.dateIso) },
+    vars: { user: '', reviewedDate: new Date() },
```
Comment (mathjazz) [inline translate/src/modules/history/components/HistoryTranslation.tsx:199]: Can we completely drop `reviewedDate` here?

File: translate/public/locale/en-US/translate.ftl (hunk 3 above)
Comment (harmitgoswami) [reply translate/public/locale/en-US/translate.ftl:372]: This change allows me to print the date without the Weekday as well. Is there a more succinct way of doing this?

File: translate/src/modules/history/components/HistoryTranslation.tsx (hunk 4 above)
Comment (harmitgoswami) [reply translate/src/modules/history/components/HistoryTranslation.tsx:199]: Potentially... is there a way for me to pass `approvedDate` or `rejectedDate` to the .ftl file if we do remove this variable?

File: translate/public/locale/en-US/translate.ftl (hunk 3 above)
Comment (mathjazz) [reply translate/public/locale/en-US/translate.ftl:372]: This format looks OK. It's the tooltip shown when you hover the translation creation date that is showing the weekday:
https://github.com/mozilla/pontoon/issues/2763#issuecomment-2203002727

File: translate/src/modules/history/components/HistoryTranslation.tsx (hunk 4 above)
Comment (mathjazz) [reply translate/src/modules/history/components/HistoryTranslation.tsx:199]: Ah, I missed that this was part of an FTL message.

File: pontoon/base/views.py [hunk 5]
```diff
@@ -418,7 +418,9 @@ def get_translation_history(request):
                 "date": t.date.strftime("%b %d, %Y %H:%M"),
                 "date_iso": t.date.isoformat(),
```
Comment (mathjazz) [inline pontoon/base/views.py:419]: We no longer use `translation.date` on frontend, so we can drop it. We can also drop `.isoformat()`. At that point we should rename `date_iso` to `date` (also on the frontend).

```suggestion
                "date": t.date,
```

File: translate/src/modules/history/components/HistoryTranslation.tsx [hunk 6]
```diff
@@ -227,6 +233,17 @@ export function HistoryTranslationBase({
...
   let canComment = user.isAuthenticated;
 
+  const customDateFormat = (date: Date) => {
+    return new Intl.DateTimeFormat('en', {
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric',
+      hour: 'numeric',
+      minute: 'numeric',
+      second: 'numeric',
```
Comment (mathjazz) [inline translate/src/modules/history/components/HistoryTranslation.tsx:243]: To be consistent with the format used for the review timestamp and more concise, let's use this:
```suggestion
      dateStyle: 'long',
      timeStyle: 'medium',
```

File: translate/src/modules/history/components/HistoryTranslation.tsx (hunk 4 above)
Comment (mathjazz) [reply translate/src/modules/history/components/HistoryTranslation.tsx:199]: Can we use `''` instead of `new Date()`?

Comment (harmitgoswami) [reply translate/src/modules/history/components/HistoryTranslation.tsx:199]: I seem to get a type error if I try to switch to `''`. Should I still try to move forward with using `''` instead of `new Date()`?

Screenshot 2024-07-03 at 1 53 47 PM

Comment (mathjazz) [reply translate/src/modules/history/components/HistoryTranslation.tsx:199]: Makes sense. Let's keep `new Date()`.

File: translate/src/modules/history/components/HistoryTranslation.tsx [hunk 7]
```diff
@@ -282,8 +295,9 @@ export function HistoryTranslationBase({
                 </Localized>
                 <ReactTimeAgo
                   dir='ltr'
-                  date={new Date(translation.dateIso)}
-                  title={`${translation.date} UTC`}
+                  date={new Date(translation.date)}
+                  title=''
```
Comment (mathjazz) [inline translate/src/modules/history/components/HistoryTranslation.tsx:299]: Seems like we can remove the `title` attribute?

**Review by mathjazz — APPROVED** [review]
Nice work! Only one comment left.
